Sources/web3swift/Tokens/ERC1376/Web3+ERC1376.swift

```diff
@@ -66,20 +66,15 @@ protocol IERC1376: IERC20 {
 }
 
 // FIXME: Rewrite this to CodableTransaction
-public class ERC1376: IERC1376, ERC20BaseProperties {
-
-    internal var _name: String?
-    internal var _symbol: String?
-    internal var _decimals: UInt8?
-    internal var _hasReadProperties: Bool = false
-
+public class ERC1376: IERC1376, ERCBaseProperties {
+    public var basePropertiesProvder: ERCBasePropertiesProvider
```

Comment:
1) Typo in the name `basePropertiesProvder`. The `..Provder` missing `i`;
2) We cannot name the class `ERCBasePropertiesProvider` as it is not related to all ERCs. The properties managed by this class are only related to ERC-20 and all other ERCs that inherit the ERC-20. And besides that just adding perfix `ERC` is strange as `ERC` means Ethereum Request for Comments and it doesn't make sense here. Only with a number `ERC` prefix will be useful.

Reply:
Same goes for `ERCBaseProperties` naming.

Reply:
Ah yea I was actually wondering about whether they actually inherit from ERC-20 but didn't look it up. Thanks for claryfying.